application/common/tests/test_dynamodb.py

```diff
@@ -46,3 +64,108 @@ def test_add_change_request_to_dynamodb(dynamodb_table_create, change_event, dyn
+        ExpressionAttributeValues={
+            ":odscode": {"S": change_event["ODSCode"]},
+        },
+        ScanIndexForward=False,
+    )
+    assert resp.get("Count") == 1
+    latest_sequence_number = get_latest_sequence_id_for_a_given_odscode_from_dynamodb(change_event["ODSCode"])
+    assert latest_sequence_number == 20
+
+
+def test_no_records_in_db_for_a_given_odscode(dynamodb_table_create, change_event):
+    latest_sequence_number = get_latest_sequence_id_for_a_given_odscode_from_dynamodb(change_event["ODSCode"])
+    assert latest_sequence_number == 0
+
+
+@patch.object(Logger, "error")
+def test_get_latest_sequence_id_for_different_change_event_from_dynamodb(
+    mock_logger, dynamodb_table_create, change_event, dynamodb_client
+):
+    event_received_time = int(time())
+    odscode = change_event["ODSCode"]
+    cevent = change_event.copy()
+    add_change_request_to_dynamodb(cevent, 1, str(event_received_time))
+    add_change_request_to_dynamodb(random_change_event(cevent), 2, str(event_received_time))
+    add_change_request_to_dynamodb(random_change_event(cevent), 3, str(event_received_time))
+    add_change_request_to_dynamodb(random_change_event(cevent), 4, str(event_received_time))
+    add_change_request_to_dynamodb(random_change_event(cevent), 44, str(event_received_time))
+    resp = dynamodb_client.query(
+        TableName=environ["CHANGE_EVENTS_TABLE_NAME"],
+        IndexName="gsi_ods_sequence",
+        KeyConditionExpression="ODSCode = :odscode",
+        ExpressionAttributeValues={
+            ":odscode": {"S": odscode},
+        },
+        ScanIndexForward=False,
+        ProjectionExpression="ODSCode,SequenceNumber",
+    )
+    expected_latest_sequence_number = 44
+    expected_count = 5
+    expected_items = [
+        {
+            "ODSCode": {
+                "S": "TES73",
+            },
+            "SequenceNumber": {
+                "N": "44",
+            },
+        },
+        {
+            "ODSCode": {
+                "S": "TES73",
+            },
+            "SequenceNumber": {
+                "N": "4",
+            },
+        },
+        {
+            "ODSCode": {
+                "S": "TES73",
+            },
+            "SequenceNumber": {
+                "N": "3",
+            },
+        },
+        {
+            "ODSCode": {
+                "S": "TES73",
+            },
+            "SequenceNumber": {
+                "N": "2",
+            },
+        },
+        {
+            "ODSCode": {
+                "S": "TES73",
+            },
+            "SequenceNumber": {
+                "N": "1",
+            },
+        },
+    ]
+    assert resp.get("Items") == expected_items
+    assert resp.get("Count") == expected_count
+    latest_sequence_number = get_latest_sequence_id_for_a_given_odscode_from_dynamodb(odscode)
+    assert latest_sequence_number == expected_latest_sequence_number
+
+
+def random_change_event(ce):
+    ce["dummy_data"] = str(time())
```

Comment:
wondering if instead of adding a random field and using something that changes with each run such as time, could we just create a copy and change a field e.g

```python 
def copy_and_modify_website(ce, new_website:str):
  copy = ce.copy()
  copy["Contacts"][0]["ContactValue"] = new_website
  return copy
```
Then call it with explicit values in the test
e.g.
```python
 add_change_request_to_dynamodb(cevent.copy(), 1, str(event_received_time))
    add_change_request_to_dynamodb(copy_and_modify_website(cevent, "www.test1.com"), 2, str(event_received_time))
    add_change_request_to_dynamodb(copy_and_modify_website(cevent, "www.test2.com"), 3, str(event_received_time))
    add_change_request_to_dynamodb(copy_and_modify_website(cevent, "www.test3.com"), 4, str(event_received_time))
    add_change_request_to_dynamodb(copy_and_modify_website(cevent, "www.test4.com"), 44, str(event_received_time))
```

Reply:
much better solution :)